modules/primo/src/Primo/Ting/PrimoStatementRenderer.php

```diff
@@ -0,0 +1,191 @@
+          $carry = array_merge($carry, array_map(function ($statement) {
+            return new BooleanStatementGroup([$statement]);
+          }, $group_statements));
+        }
+        else {
+          // The group referenced more than one field and was not joined by AND.
+          throw new UnsupportedSearchQueryException("Primo only supports AND between different field.");
+        }
+      }
+      else {
+        // The all statements in the group is referencing the same field, go
+        // ahead and add it.
+        $carry[] = $statement;
+      }
+      return $carry;
+    }, []);
+  }
+
+  /**
+   * Render a group into a statement.
+   *
+   * The groups must be "Primo" compatible. See parameter documentation for
+   * details.
+   *
+   * @param \Ting\Search\BooleanStatementGroup $group
+   *   A list of BooleanStatementGroup instances containing one or more
+   *   statements all against the same field if they are joined by OR.
+   *
+   * @return array
+   *   An associative array with keys that are valid primo query parameters such
+   *   as "query" and values that are an array of valid primo parameter-values.
+   *   Eg. ['query' => ['author,exact,Rowlings', 'title,exact,Harry Potter']].
+   */
+  protected function renderGroup(BooleanStatementGroup $group) {
+    // We're a group group containing one or more field statements (against the
+    // same field) into a single query=<fieldname>,exact,<list of values>.
+    // See https://developers.exlibrisgroup.com/primo/apis/webservices/xservices/search/briefsearch
+    // for more details.
+    $statements = $group->getStatements();
+
+    // The group has been preprocessed so we know it to contain at least one
+    // statement and all statements will be against the same field so the
+    // following is safe.
+    $field_name = $statements[0]->getName();
+    // If this is a field we have a mapping for, map it.
+    if (isset($this->mapping[$field_name])) {
+      $field_name = $this->mapping[$field_name];
+    }
+    $values = array_map(function (TingSearchFieldFilter $statement) {
+      // "Escape" the string by replacing commas with spaces.
+      return str_replace(',', ' ', $this->escapeValue($statement->getValue()));
+    }, $statements);
+
+    $values_joined = implode(' ' . $group->getLogicOperator() . ' ', $values);
+
+    // Eg. query=rtype,exact,audiobooks OR articles.
+    return ['query' => [$field_name . ',exact,' . $values_joined]];
+  }
+
+  /**
+   * Escapes a value to be placed into a primo query parameter.
+   *
+   * @param string $value
+   *   The value.
+   *
+   * @return string
+   *   The escaped string.
+   */
+  public function escapeValue($value) {
+    $replacements = [
+      // AND and OR are keywords, strip them.
+      ' AND ' => ' ',
+      ' OR ' => ' ',
+      // Commas should be replaced with spaces.
+      ',' => ' ',
+    ];
+
+    return str_replace(array_keys($replacements), array_values($replacements), $value);
```

Comment:
If we already have a map from search to replace then we can use that with [`strtr`](http://php.net/manual/en/function.strtr.php).